pkg/annotations/ingress_annotations.go

```diff
@@ -108,6 +129,23 @@ func UsePrivateIP(ing *v1beta1.Ingress) (bool, error) {
 	return parseBool(ing, UsePrivateIPKey)
 }
 
+// BackendProtocol provides value for protocol to be used with the backend
+func BackendProtocol(ing *v1beta1.Ingress) (ProtocolEnum, error) {
+	protocol, err := parseString(ing, BackendProtocolKey)
+	if err != nil {
+		return HTTP, err
+	}
+
+	protocolLowerCase := strings.ToLower(protocol)
+	for protocolEnum, mappedProtocol := range ProtocolEnumLookup {
```

Comment:
It seems that you are using the map al a list of tuples. It's a fine approach. But since we have a map we could turn this from ON to O1 lookup.

What if we inversed the map above to: `map[string]ProtocolEnum{"http": HTTP, "https": HTTPS}`
then here we could do `protocolEnum, ok := ProtocolEnumLookup[protocolLowerCase]`

Reply:
thanks!